sdk/python/kfp/components/_python_op.py

```diff
@@ -694,6 +695,118 @@ def add_multiply_two_numbers(a: float, b: float) -> NamedTuple('DummyName', [('s
     return _create_task_factory_from_component_spec(component_spec)
 
 
+def create_component_from_func(
+    func: Callable,
+    output_component_file: str=None,
+    base_image: str = None,
+    packages_to_install: List[str] = None,
+):
+    '''
+    Converts a Python function to a component and returns a task factory (a function that accepts arguments and returns a task object).
+
+    Function name and docstring are used as component name and description.
```

Comment:
Sorry I forgot how do we deal with name conflict here. Do we make them unique by appending index or raise compile time error?

Reply:
At the single component level, there can be no conflicts - the component name can be anything. (We cannot prevent different component authors from using the same names.)

When serializing the pipeline graph, however, there is a need to give different IDs to different tasks. We could generate them [randomly](https://github.com/kubeflow/pipelines/blob/469e7196cfdf4eca851cb9f2c9e96197067c4c2d/sdk/python/kfp/dsl/_container_op.py#L666) or sequentially or using hash function, but at this moment the DSL compiler sets the task id to component name + optional sequential number that makes the ids unique. https://github.com/kubeflow/pipelines/blob/469e7196cfdf4eca851cb9f2c9e96197067c4c2d/sdk/python/kfp/dsl/_pipeline.py#L218